### DISCUSSION TITLE
"Objcopy: error: the input file '*.dll' is empty" when running the build script

### DISCUSSION THREAD

**Discussion by P-Mousavi** [tianocore/edk2#5987] [Q&A]
Greetings, I've been trying to build the edk2 from source in a Kali 2023.3 VM, followed every step from [the linux guide](https://github.com/tianocore/tianocore.github.io/wiki/Common-instructions), but I got a few error messages just like the one in the title for some files like "**MnpDxe.dll**" each time I run the "build" command.

"**Make -C BaseTools**" works just fine btw, the GCC version is **13.1**, python **3.11** and the config is "**TARGET_ARCH=X64, TOOL_CHAIN_TAG=GCC5, ACTIVE_PLATFORM=MdeModulePkg/MdeModulePkg.dsc**". Also tried the "EmulatorPkg" platform, no luck (I'm specifically after the emulator to use alongside QEMU).

Any ideas?

**Comment by r1chard-lyu** [comment]
Could you please verify that all required dependencies are installed? To help diagnose the issue, please follow these steps. Once we have this information, we can determine whether it's a compilation error, configuration issue, or file permissions problem. This will help provide more targeted assistance.

1. Confirm all dependencies are properly installed:
```bash
sudo apt update
sudo apt install build-essential uuid-dev iasl git gcc nasm python3-distutils
```
2. Verify that the BaseTools are compiled correctly:
```bash
make -C BaseTools
```
3. Try the build command again:
```bash
source edksetup.sh
make -C BaseTools
build -a X64 -t GCC5 -p MdeModulePkg/MdeModulePkg.dsc
```
4. Review Build Output and Post Error Messages:
Analyze the build log for any error or warning messages. These messages can pinpoint where the build process is failing or encountering issues. If you encounter any errors, please copy and paste the complete error messages here, including as much context as possible to assist with troubleshooting.

**Reply by P-Mousavi** [reply]
Thanks for the reply. I've all the dependencies you mentioned installed except gcc-5 which is quite old and probably the only way to add it to my toolchain right now is to compile it from source which is opening another can of worms. Might try to install an older version of debian someday to test it.

**Reply by r1chard-lyu** [reply]
@P-Mousavi For EDK2, as long as the GCC version installed on your operating system is >= 5, the build option should be -t GCC5. Therefore, it's perfectly fine to use a higher version of GCC. Has this issue been resolved? Are there any further related questions? If the issue has been resolved, I would recommend closing this discussion.
